openfe/tests/protocols/openmm_septop/test_septop_protocol.py

```diff
@@ -0,0 +1,1261 @@
+# This code is part of OpenFE and is licensed under the MIT license.
+# For details, see https://github.com/OpenFreeEnergy/openfe
+import pathlib
+import pytest
+import math
+
+import openfe.protocols.openmm_septop
+from openfe import ChemicalSystem, SolventComponent
+from openfe.protocols.openmm_septop import (
+    SepTopProtocol,
+    SepTopComplexSetupUnit,
+    SepTopComplexRunUnit,
+    SepTopSolventSetupUnit,
+    SepTopSolventRunUnit,
+    SepTopProtocolResult,
+)
+from openfe.protocols.openmm_septop.equil_septop_method import _check_alchemical_charge_difference
+from openfe.protocols.openmm_utils import system_validation
+from openfe.protocols.openmm_septop.utils import deserialize
+from openfe.tests.protocols.conftest import compute_energy
+
+import numpy as np
+from numpy.testing import assert_allclose
+from math import sqrt
+
+import openmm
+import openmm.app
+import openmm.unit
+from openmmtools.multistate.multistatesampler import MultiStateSampler
+from openff.units import unit as offunit
+import gufe
+from unittest import mock
+import json
+import mdtraj as md
+import itertools
+
+from openmmtools.alchemy import AlchemicalRegion, AbsoluteAlchemicalFactory
+from openff.units.openmm import ensure_quantity, from_openmm
+from openmm import (
+    app, XmlSerializer, MonteCarloBarostat,
+    NonbondedForce, CustomNonbondedForce
+)
+from importlib import resources
+
+E_CHARGE = 1.602176634e-19 * openmm.unit.coulomb
+EPSILON0 = (
+    1e-6
+    * 8.8541878128e-12
+    / (openmm.unit.AVOGADRO_CONSTANT_NA * E_CHARGE**2)
+    * openmm.unit.farad
+    / openmm.unit.meter
+)
+ONE_4PI_EPS0 = 1 / (4 * np.pi * EPSILON0) * EPSILON0.unit * 10.0  # nm -> angstrom
+
+
+@pytest.fixture()
+def default_settings():
+    return SepTopProtocol.default_settings()
+
+
+def test_create_default_settings():
+    settings = SepTopProtocol.default_settings()
+    assert settings
+
+
+@pytest.mark.parametrize('val', [
+    {'elec': [0.0, -1], 'vdw': [0.0, 1.0], 'restraints': [0.0, 1.0]},
+    {'elec': [0.0, 1.5], 'vdw': [0.0, 1.5], 'restraints': [-0.1, 1.0]}
+])
+def test_incorrect_window_settings(val, default_settings):
+    errmsg = "Lambda windows must be between 0 and 1."
+    lambda_settings = default_settings.complex_lambda_settings
+    with pytest.raises(ValueError, match=errmsg):
+        lambda_settings.lambda_elec_A = val['elec']
+        lambda_settings.lambda_vdw_A = val['vdw']
+        lambda_settings.lambda_restraints_A = val['restraints']
```

Comment:
I think the cases on lines 75 and 76 need to be worked into the parameterize list, when the error is raised on 74 it wont test the other cases.